packages/nouns-webapp/src/components/NavWallet/index.tsx

```diff
@@ -0,0 +1,238 @@
+
+  const [buttonUp, setButtonUp] = useState(false);
+  const [showConnectModal, setShowConnectModal] = useState(false);
+  const history = useHistory();
+  const { library: provider } = useEthers();
+  const activeAccount = useAppSelector(state => state.account.activeAccount);
+  const { deactivate } = useEthers();
+  const ens = useReverseENSLookUp(address);
+  const shortAddress = useShortAddress(address);
+
+  const setModalStateHandler = (state: boolean) => {
+    setShowConnectModal(state);
+  };
+
+  const switchWalletHandler = () => {
+    setShowConnectModal(false);
+    setButtonUp(false);
+    deactivate();
+    setShowConnectModal(false);
+    setShowConnectModal(true);
+  };
+
+  const disconectWalletHandler = () => {
+    setShowConnectModal(false);
+    setButtonUp(false);
+    deactivate();
+  };
+
+  const statePrimaryButtonClass = usePickByState(
+    classes.whiteInfo,
+    classes.coolInfo,
+    classes.warmInfo,
+    history,
+  );
+
+  const stateSelectedDropdownClass = usePickByState(
+    classes.whiteInfoSelected,
+    classes.dropdownActive,
+    classes.dropdownActive,
+    history,
+  );
+
+  const mobileTextColor = usePickByState(
+    'rgba(140, 141, 146, 1)',
+    'rgba(121, 128, 156, 1)',
+    'rgba(142, 129, 127, 1)',
+    history,
+  );
+
+  const mobileBorderColor = usePickByState(
+    'rgba(140, 141, 146, .5)',
+    'rgba(121, 128, 156, .5)',
+    'rgba(142, 129, 127, .5)',
+    history,
+  );
+
+  const connectWalletButtonStyle = usePickByState(
+    NavBarButtonStyle.WHITE_WALLET,
+    NavBarButtonStyle.COOL_WALLET,
+    NavBarButtonStyle.WARM_WALLET,
+    history,
+  );
+
+  const customDropdownToggle = React.forwardRef<RefType, Props>(({ onClick, value }, ref) => (
+    <>
+      <div
+        className={clsx(
+          classes.wrapper,
+          buttonUp ? stateSelectedDropdownClass : statePrimaryButtonClass,
+        )}
+        onClick={e => {
+          e.preventDefault();
+          onClick(e);
+        }}
+      >
+        <div className={classes.button}>
+          <div className={classes.icon}>
+            {' '}
+            <Davatar size={21} address={address} provider={provider} />
+          </div>
+          <div className={classes.address}>{ens ? ens : shortAddress}</div>
+          <div className={buttonUp ? classes.arrowUp : classes.arrowDown}>
+            <FontAwesomeIcon icon={buttonUp ? faSortUp : faSortDown} />{' '}
+          </div>
+        </div>
+      </div>
+    </>
+  ));
+
+  const CustomMenu = React.forwardRef((props: CustomMenuProps, ref: React.Ref<HTMLDivElement>) => {
+    return (
+      <div
+        ref={ref}
+        style={props.style}
+        className={props.className}
+        aria-labelledby={props.labeledBy}
+      >
+        <div>
+          <div
+            onClick={switchWalletHandler}
+            className={clsx(
+              classes.dropDownTop,
+              classes.button,
+              classes.switchWalletText,
+              usePickByState(
```

Comment:
"Helper functions that contain hooks as parameters to functions in props" is a mouthful, which is a good sign it's probably a smell as well.

Also not blocking but could use some attention.